tests/window_management_tests/test_minimal_window_manager.cpp

```diff
@@ -554,40 +558,152 @@ TEST_F(MinimalWindowManagerTest, can_resize_south_east_with_touch)
     EXPECT_EQ(window.size(), geom::Size(50, 50));
 }
 
-TEST_F(MinimalWindowManagerTest,  new_windows_are_inserted_above_older_windows)
+template <uint N>
+class MultipleWindowsMinimalWindowManagerTest : public MinimalWindowManagerTest
+{
+public:
+    void SetUp() override
+    {
+        MinimalWindowManagerTest::SetUp();
+        miral::WindowSpecification spec;
+        spec.size() = { geom::Width {100}, geom::Height{100} };
+        spec.depth_layer() = mir_depth_layer_application;
+
+        for (uint i = 0; i < N; i++)
+        {
+            apps[i] = open_application("app" + i);
+            windows[i] = create_window(apps[i], spec);
+        }
+    }
+
+    miral::Application apps[N];
+    miral::Window windows[N];
+};
+
+using TwoWindowsMinimalWindowManagerTest = MultipleWindowsMinimalWindowManagerTest<2>;
+using ThreeWindowsMinimalWindowManagerTest = MultipleWindowsMinimalWindowManagerTest<3>;
```

Comment:
I know it reads easier, but selecting tests with globbing is easier, and more consistent with prior art, with `--gtest_filter=MinimalWindowManager*`
```suggestion
using MinimalWindowManagerWithTwoWindows = MultipleWindowsMinimalWindowManagerTest<2>;
using MinimalWindowManagerWithThreeWindows = MultipleWindowsMinimalWindowManagerTest<3>;
```